pkg/validator/sarif.go

```diff
@@ -0,0 +1,49 @@
+package validator
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/xeipuuv/gojsonschema"
+)
+
+type SarifValidator struct{}
+
+func (SarifValidator) Validate(b []byte) (bool, error) {
+	var report map[string]interface{}
+	err := json.Unmarshal(b, &report)
+	if err != nil {
+		customErr := getCustomErr(b, err)
+		return false, customErr
+	}
+
+	schemaUrl, ok := report["$schema"]
+	if !ok {
+		return false, errors.New("error - no schema specified")
+	}
+
+	loadedSchema := gojsonschema.NewReferenceLoader(schemaUrl.(string))
+	loadedReport := gojsonschema.NewRawLoader(report)
+
+	schema, err := gojsonschema.NewSchema(loadedSchema)
+	if err != nil {
+		return false, errors.New("error - schema isn't valid")
+	}
+	result, err := schema.Validate(loadedReport)
+	if err != nil {
+		return false, errors.New("error - couldn't validate a report")
+	}
+	if !result.Valid() {
+		return false, formatError(result.Errors())
+	}
+	return true, nil
+}
+
+func formatError(resultEerrors []gojsonschema.ResultError) error {
+	var errorDescription string
+	for _, err := range resultEerrors {
+		errorDescription = err.Description()
+	}
+	return fmt.Errorf("error - %s", errorDescription)
+}
```

Comment:
```suggestion
func formatError(resultErrors []gojsonschema.ResultError) error {
	var errorDescription string
	for _, err := range resultErrors {
		errorDescription = err.Description()
	}
	return fmt.Errorf("error - %s", errorDescription)
}
```

But I don't get it.

You only have one error? then I would use this

```suggestion
func formatError(resultErrors []gojsonschema.ResultError) error {
	errorDescription := "unknown"
	for _, err := range resultErrors {
		errorDescription = err.Description()
		break
	}
	return fmt.Errorf("error - %s", errorDescription)
}
```

Otherwise there is something strange with the loop, as only the last one is reported

```suggestion
func formatError(resultErrors []gojsonschema.ResultError) error {
	var errorDescription []string
	for _, err := range resultErrors {
		errorDescription = append(errorDescription, err.Description())
	}
	return fmt.Errorf("error - %s", strings.Join(", ", errorDescription))
}
```

Reply:
Yeah, I've tried to achieve the same logic (e.g.: if there are multiple validation errors, show them all, instead of the last one from the slice). But ended up in showing only one.

Made changes, so that if there are multiple errors, show them all with a `,` separator.

Thank you @ccoVeille for taking time to look at the PR.